### PR TITLE
Show Course and Unit Rollups

### PR DESCRIPTION
**Pull request by dmcavoy**
Hooks up the react components for Course and Unit Rollup to show on the actual pages!

Screen Shot 2021-03-25 at 9 12 07 AM

Screen Shot 2021-03-25 at 9 12 51 AM

## Links

- jira ticket: [PLAT-752](https://codedotorg.atlassian.net/browse/PLAT-752)
- jira ticket: [PLAT-704](https://codedotorg.atlassian.net/browse/PLAT-704)
- jira ticket: [PLAT-398](https://codedotorg.atlassian.net/browse/PLAT-398)
- jira ticket: [PLAT-642](https://codedotorg.atlassian.net/browse/PLAT-642)

## Testing story

- Tested each page manually for CSD 2021 and CSD 3 2021

## Follow-up work

- Adding links to these pages from Course And Unit Overview pages
- Improvements to these pages based on feedback

## PR Checklist:

- [x] Tests provide adequate coverage
- [x] Privacy and Security impacts have been assessed
- [x] Code is well-commented
- [x] New features are translatable or updates will not break translations
- [x] Relevant documentation has been added or updated
- [x] User impact is well-understood and desirable
- [x] Pull Request is labeled appropriately
- [x] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/app/controllers/courses_controller.rb [hunk 1]
```diff
@@ -112,27 +112,31 @@ def vocab
     raise ActiveRecord::RecordNotFound unless unit_group
     # Assumes if one unit in a unit group is migrated they all are
     return render :forbidden unless unit_group.default_scripts[0].is_migrated
+    render 'vocab', locals: {course_summary: unit_group.summarize_for_rollup(@current_user)}
```
Comment (davidsbailey) [inline dashboard/app/controllers/courses_controller.rb]: I don't see this pattern very often in our codebase. please set a variable like this instead:
```
@course_summary = unit_group.summarize_for_rollup(@current_user)
```
and then access `@course_summary` from the haml file. then you will not need to explicitly `render 'vocab'`. same for other controller methods in this file.

File: dashboard/app/controllers/scripts_controller.rb [hunk 2]
```diff
@@ -145,18 +145,22 @@ def instructions
 
   def vocab
     return render :forbidden unless can? :read, @script
+    render 'vocab', locals: {unit_summary: @script.summarize_for_rollup(@current_user)}
```
Comment (davidsbailey) [inline dashboard/app/controllers/scripts_controller.rb]: please either set `@unit_summary` here, or skip entirely and just call `@script.summarize_for_rollup(@current_user)` from the haml file

File: dashboard/app/models/script.rb [hunk 3]
```diff
@@ -1465,6 +1465,22 @@ def summarize(include_lessons = true, user = nil, include_bonus_levels = false)
...
 
+  def summarize_for_rollup(user = nil)
+    summary = {
+      title: title_for_display,
+      name: name,
+      link: script_path(self)
+    }
+
+    # Filter out stages that have a visible_after date in the future
+    filtered_lessons = lessons.select {|lesson| lesson.published?(user)}
```
Comment (davidsbailey) [inline dashboard/app/models/script.rb:1476]: nice catch!

**Review by davidsbailey — APPROVED** [review]
Nice work @dmcavoy ! It's awesome to see this come together!

**Comment by dmcavoy** [comment]
Rebased to staging while tests run.

**Pull request merged**